include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //uint32_t              maxBoundDescriptorSets;
+    //uint32_t              maxPerStageDescriptorSamplers;
+    //uint32_t              maxPerStageDescriptorUniformBuffers;
+    uint32_t maxPerStageDescriptorSSBOs;
+    //uint32_t              maxPerStageDescriptorSampledImages;
+    //uint32_t              maxPerStageDescriptorStorageImages;
+    //uint32_t              maxPerStageDescriptorInputAttachments;
+    //uint32_t              maxPerStageResources;
+    //uint32_t              maxDescriptorSetSamplers;
+    uint32_t maxDescriptorSetUBOs;
+    uint32_t maxDescriptorSetDynamicOffsetUBOs;
+    uint32_t maxDescriptorSetSSBOs;
+    uint32_t maxDescriptorSetDynamicOffsetSSBOs;
+    uint32_t maxDescriptorSetImages;
+    uint32_t maxDescriptorSetStorageImages;
+    //uint32_t              maxDescriptorSetInputAttachments;
+    //uint32_t              maxVertexInputAttributes;
+    //uint32_t              maxVertexInputBindings;
+    //uint32_t              maxVertexInputAttributeOffset;
+    //uint32_t              maxVertexInputBindingStride;
+    //uint32_t              maxVertexOutputComponents;
+    //uint32_t              maxTessellationGenerationLevel;
+    //uint32_t              maxTessellationPatchSize;
+    //uint32_t              maxTessellationControlPerVertexInputComponents;
+    //uint32_t              maxTessellationControlPerVertexOutputComponents;
+    //uint32_t              maxTessellationControlPerPatchOutputComponents;
+    //uint32_t              maxTessellationControlTotalOutputComponents;
+    //uint32_t              maxTessellationEvaluationInputComponents;
+    //uint32_t              maxTessellationEvaluationOutputComponents;
+    //uint32_t              maxGeometryShaderInvocations;
+    //uint32_t              maxGeometryInputComponents;
+    //uint32_t              maxGeometryOutputComponents;
+    //uint32_t              maxGeometryOutputVertices;
+    //uint32_t              maxGeometryTotalOutputComponents;
+    //uint32_t              maxFragmentInputComponents;
+    //uint32_t              maxFragmentOutputAttachments;
+    //uint32_t              maxFragmentDualSrcAttachments;
+    //uint32_t              maxFragmentCombinedOutputResources;
+    uint32_t maxComputeSharedMemorySize;
+    //uint32_t              maxComputeWorkGroupCount[3];
+    //uint32_t              maxComputeWorkGroupInvocations;
+    uint32_t maxWorkgroupSize[3];
+    //uint32_t              subPixelPrecisionBits;
+    //uint32_t              subTexelPrecisionBits;
+    //uint32_t              mipmapPrecisionBits;
+    //uint32_t              maxDrawIndexedIndexValue;
+    uint32_t maxDrawIndirectCount;
+    //float                 maxSamplerLodBias;
+    float    maxSamplerAnisotropyLog2;
+    uint32_t maxViewports;
+    uint32_t maxViewportDims[2];
+    //float                 viewportBoundsRange[2];
+    //uint32_t              viewportSubPixelBits;
+    //size_t                minMemoryMapAlignment;
```

Comment:
Do we want this?
> minMemoryMapAlignment is the minimum required alignment, in bytes, of host visible memory allocations within the host address space. When mapping a memory allocation with [vkMapMemory](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkMapMemory.html), subtracting offset bytes from the returned pointer will always produce an integer multiple of this limit.

Reply:
yes, it exists as `GL_MIN_MAP_BUFFER_ALIGNMENT` under desktop GL

Reply:
Done,
 but how did you figure out `GL_MIN_MAP_BUFFER_ALIGNMENT` doesn't exist in GLES? and how should I figure If I have to protect some `GetIntegerv`s with `isGLES` ?

Did you refer to these pages and saw that only GL has `GL_MIN_MAP_BUFFER_ALIGNMENT `?
https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glMapBufferRange.xhtml
https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/glMapBufferRange.xhtml

Does this mean for every glGetIntegerV I do I should check the two reference pages and see if both have it or only one?

Reply:
> but how did you figure out `GL_MIN_MAP_BUFFER_ALIGNMENT` doesn't exist in GLES? and how should I figure If I have to protect some `GetIntegerv`s with `isGLES` ?

I looked up the documentation for `glGet` under GLES **and then I also looked up the `EXT_buffer_storage` GLES extension spec to see if it defines the constant**

> Does this mean for every glGetIntegerV I do I should check the two reference pages and see which one has that?

Basically that, but you need to be very careful which documentation you check, sometimes there's shit missing, so check 4.6 GL and 3.2 GLES